airgun/entities/subscription.py

```diff
@@ -0,0 +1,180 @@
+from selenium.common.exceptions import NoSuchElementException
+
+from navmazing import NavigateToSibling
+from wait_for import wait_for
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.views.subscription import (
+    SubscriptionListView,
+    ManageManifestView,
+    DeleteManifestConfirmationView,
+    AddSubscriptionView,
+    SubscriptionDetailsView,
+)
+
+
+class SubscriptionEntity(BaseEntity):
+    def _wait_for_process_to_finish(self, has_manifest=False):
+        view = self.navigate_to(self, 'All')
+        view.progressbar.wait_displayed(timeout=20)
+        wait_for(
+                lambda: not view.progressbar.is_displayed,
+                timeout=60*10,
+                logger=view.progressbar.logger
+        )
+        wait_for(
+                lambda: self.has_manifest == has_manifest,
+                timeout=10,
+                logger=view.progressbar.logger
+        )
+        view.flash.dismiss()
+
+    @property
+    def has_manifest(self):
+        view = self.navigate_to(self, 'All')
+        return "disabled" not in view.browser.classes(view.add_button)
+
+    def add_manifest(self, manifest_file):
+        view = self.navigate_to(self, 'Manage Manifest')
+        view.fill({
+            'manifest.manifest_file': manifest_file,
+        })
+        self._wait_for_process_to_finish(has_manifest=True)
+
+    def delete_manifest(self, really=True):
+        view = self.navigate_to(self, 'Delete Manifest Confirmation')
+        if not really:
```

Comment:
please remove:
```
        if not really:
            view.cancel_button.click()
            try:
                while view.is_displayed:
                    pass
            except NoSuchElementException:
                pass
            self.navigate_to(self, 'Manage Manifest').close_button.click()
            return
```
as we will never verify that functionality for tier2-tier4 tests